yt_dlp/extractor/cda.py

```diff
@@ -351,3 +351,63 @@ def extract_format(page, version):
             extract_format(webpage, resolution)
 
         return merge_dicts(info_dict, info)
+
+
+class CDAFolderIE(InfoExtractor):
+    _MAX_PAGE_NUMBER = 200
+    _VALID_URL = r'https?://(?:www\.)?cda\.pl/[0-9a-zA-Z_]+/folder/(?P<id>[0-9a-z]+)(/vfilm)?'
+    _TESTS = [
+        {
+            'url': 'https://www.cda.pl/domino264/folder/31188385',
+            'info_dict': {
+                'id': '31188385',
+                'title': 'SERIA DRUGA',
+            },
+            'playlist_count': 13,
+        },
+        {
+            'url': 'https://www.cda.pl/smiechawaTV/folder/2664592/vfilm',
+            'info_dict': {
+                'id': '2664592',
+                'title': 'VideoDowcipy - wszystkie odcinki',
+            },
+            'playlist_count': 71,
+        },
+        {
+            'url': 'https://www.cda.pl/DeliciousBeauty/folder/19129979/vfilm',
+            'info_dict': {
+                'id': '19129979',
+                'title': 'TESTY KOSMETYKÓW',
+            },
+            'playlist_count': 139,
+        }]
+
+    def _real_extract(self, url):
+        folder_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, folder_id)
+
+        title = self._html_search_meta('og:title', webpage, 'title')
+
+        def extract_page_entries(url):
+            webpage = self._download_webpage(url, folder_id, f'Extracting videos list from {url}')
+            items = re.findall(r'<a[^>]+href="/video/([0-9a-z]+)" class=', webpage)
+            page_entries = []
+            for video_id in items:
+                self.to_screen(f'Extracting {video_id}')
+                self.url_result(f'https://www.cda.pl{video_id}', CDAIE.ie_key())
+                page_entries.append(self.url_result(f'https://www.cda.pl/video/{video_id}', CDAIE.ie_key(), video_id))
+            # Find max page number from html:
+            # <a href="https://www.cda.pl/CHANNEL_NAME/folder/FOLDER_ID/vfilm/MAX_NUM" class="next">&gt;<
+            next_page_number = int_or_none(self._html_search_regex(
+                r'<a[^>]+href="https://www.cda.pl/[^/]+/folder/[^/]+/vfilm/(?P<page>\d+)" class="next">&gt;<',
+                webpage, 'page count', default=None))
+
+            return page_entries, next_page_number
```

Comment:
In general, I think I would rewrite this using `OnDemandPagedList`. Should simplify the code and also make it work lazily instead of having to extract all pages in advance.

Reply:
@seproDev Is it possible to force class OnDemandPagedList to start iterating from 1 instead of 0?

https://www.cda.pl/DeliciousBeauty/folder/19129979/vfilm/1 is the first page, because https://www.cda.pl/DeliciousBeauty/folder/19129979/vfilm/0 does not exist

Reply:
Done